### PR TITLE
chore: integration test small improvements

### PR DESCRIPTION
**Pull request by tomaszpowroznik**

### CODE REVIEW

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/RepresentativeSubmittedEventControllerTest.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
 @OverrideAutoConfiguration(enabled = true)
 class RepresentativeSubmittedEventControllerTest extends AbstractControllerTest {
 
-    private static final String REPPRESENTATIVE_FULLNAME = "John Smith";
+    private static final String REPRESENTATIVE_FULLNAME = "John Smith";
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/RepresentativeSubmittedEventControllerTest.java:49]: object mapper shouldn't be required in this test, is in the base class

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageLegalRepresentativeAboutToSubmitControllerTest.java [hunk 2]
```diff
@@ -32,20 +31,16 @@
...
         .email(REPRESENTATIVE_EMAIL)
         .organisation("organisation")
         .telephoneNumber("07500045455")
         .build();
     private static final Long CASE_ID = 12345L;
-    public static final String USER_ID = RandomStringUtils.randomAlphanumeric(10);
-
-    @MockBean
-    private AuthTokenGenerator authTokenGenerator;
+    private static final String USER_ID = RandomStringUtils.randomAlphanumeric(10);
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageLegalRepresentativeAboutToSubmitControllerTest.java:43]: we have a `USER_ID` in the base class

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AddCaseNumberControllerAboutToSubmitTest.java [hunk 3]
```diff
@@ -1,55 +1,46 @@
...
+            .build();
 
-        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToSubmitEvent(callbackRequest.getCaseDetails(),
-            SC_OK);
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToSubmitEvent(caseData);
 
         assertThat(callbackResponse.getErrors()).isEmpty();
-        assertThat(callbackResponse.getData()).containsKey(FAMILY_MAN_CASE_NUMBER_KEY);
-        assertThat(callbackResponse.getData().get(FAMILY_MAN_CASE_NUMBER_KEY)).isEqualTo(expectedFamilymanCaseNumber);
+        assertThat(callbackResponse.getData().get("familyManCaseNumber")).isEqualTo(expectedFamilyManCaseNumber);
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AddCaseNumberControllerAboutToSubmitTest.java:43]: better to leave like this or extract from case data?

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java [hunk 4]
```diff
@@ -43,9 +40,6 @@
     private static final Long CASE_ID = 12345L;
     private static final DocumentReference document = testDocumentReference();
 
-    @MockBean
-    private IdamClient idamClient;
-
     @MockBean
     private RequestData requestData;
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java:44]: Do you wanna do the same for request data?

Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java:44]: Also you can remove time from this one

**Review by lewisbirks — COMMENTED** [review]
Looks good, a few little things missed. Might go over them again myself once this one is merged and see if we can use case data in the place of maps in more places

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/StandardDirectionsOrderControllerAboutToStartTest.java [hunk 5]
```diff
@@ -23,12 +23,13 @@
...
 import static uk.gov.hmcts.reform.fpl.enums.ccd.fixedlists.SDORoute.UPLOAD;
 import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocumentReference;
 
 @ActiveProfiles("integration-test")
 @WebMvcTest(StandardDirectionsOrderController.class)
 @OverrideAutoConfiguration(enabled = true)
 class StandardDirectionsOrderControllerAboutToStartTest extends AbstractControllerTest {
-    private static final DocumentReference SDO = DocumentReference.builder().filename("sdo.pdf").build();
+    private static final DocumentReference SDO = testDocumentReference("sdo.pdf");
```
[the remark this answers is not in the document]
Comment (tomaszpowroznik) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/StandardDirectionsOrderControllerAboutToStartTest.java:32]: agree, did it in couple places, but it is already 40 files, as we agreed in tech retro lets keep it smaller and focused

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AddCaseNumberControllerAboutToSubmitTest.java (hunk 3 above)
Comment (tomaszpowroznik) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AddCaseNumberControllerAboutToSubmitTest.java:43]: separate PR for CaseData, not necessary thing as then we dont have test that serialisation is correct

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java (hunk 4 above)
Comment (tomaszpowroznik) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java:44]: good point. Yes i would like to have this for RequestData as well

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/RepresentativeSubmittedEventControllerTest.java (hunk 1 above)
Comment (tomaszpowroznik) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/RepresentativeSubmittedEventControllerTest.java:47]: done

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageLegalRepresentativeAboutToSubmitControllerTest.java (hunk 2 above)
Comment (tomaszpowroznik) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageLegalRepresentativeAboutToSubmitControllerTest.java]: done

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java (hunk 4 above)
Comment (tomaszpowroznik) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java]: done

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java [hunk 6]
```diff
@@ -40,9 +37,6 @@
     private static final Long CASE_ID = 12345L;
     private static final DocumentReference document = testDocumentReference();
 
-    @MockBean
-    private RequestData requestData;
-
     @Autowired
     private Time time;
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java:39]: Can be removed

**Review by lewisbirks — APPROVED** [review]

**Pull request merged**